### PR TITLE
openstack-deploy: fix list_package role - no attribut artifacts

### PR DESCRIPTION
**Pull request by djz88**

### CODE REVIEW

**Comment by djz88** [comment]
CI runs
----------
When no old jobs with artifacts found:
https://ci.suse.de/job/cloud-ardana8-job-gate-entry-scale-kvm-update-x86_64/383/
When there are old jobs with artifacts for the run:
https://ci.suse.de/job/cloud-ardana8-job-gate-entry-scale-kvm-update-x86_64/390/artifact/.artifacts/rpms/
https://ci.suse.de/job/openstack-ardana/792/artifact/.artifacts/rpms/
https://ci.suse.de/job/openstack-crowbar/1009/artifact/.artifacts/rpms/

File: scripts/jenkins/cloud/ansible/roles/list_packages/tasks/packages_from_old_jobs.yml [hunk 1]
```diff
@@ -28,66 +28,90 @@
...
 
-- name:
-  combine_files:
-    filenames1: "{{_tmpRelPathVal }}"
-    filenames2: "{{ _currentJobFileNames }}"
-  register: _all_items
+    - name: Download package lists from old jobs
+      get_url:
+        url: "https://ci.suse.de/job/{{ job_base_name }}/{{ item.0.key }}/artifact/{{ item.1.relativePath }}"
+        #url: "https://ci.suse.de/job/openstack-{{ cloud_product }}/{{ item.0.key }}/artifact/{{ item.1.relativePath }}"
```
Comment (rtamalin) [inline scripts/jenkins/cloud/ansible/roles/list_packages/tasks/packages_from_old_jobs.yml:69]: Why leave this commented out line here?

File: scripts/jenkins/cloud/ansible/roles/list_packages/tasks/packages_from_old_jobs.yml [hunk 2]
```diff
@@ -28,66 +28,90 @@
...
-    output: "{{ diff_tmp_dir }}/rpms/diffs/{{ item_virtual_hosts.jobNumber }}_{{item_virtual_hosts.currentFilename}}"
-  delegate_to: localhost
-  register: diff_files
-  loop: "{{ _all_items.combine_files }}"
-  loop_control:
-    loop_var: item_virtual_hosts
-  when: _all_items.combine_files is not none
+    - name: Set Facts - _temporary Relative Path Value from jobs
+      set_fact:
+        _tmpRelPathVal: "{{ _tmpRelPathVal + [ item.item.0.key|string + ':' + item.item.1.relativePath|basename ] }}"
```
Comment (rtamalin) [inline scripts/jenkins/cloud/ansible/roles/list_packages/tasks/packages_from_old_jobs.yml:83]: _tmpRelPathVal is not really consistent with the usual ansible fact naming scheme of all lower case with `_` delimitation of words. I'm aware that it is technically inherited from previous version of code just more of an observation

File: scripts/jenkins/cloud/ansible/roles/list_packages/tasks/packages_from_old_jobs.yml [hunk 3]
```diff
@@ -28,66 +28,90 @@
...
+      loop:   "{{ _downloaded_files.results }}"
+      loop_control:
+          label: "{{ item.item.1.relativePath }}"
+      when:
+        - (item.item.1.relativePath | dirname).endswith('rpms')
+        - (item.item.1.relativePath | basename).startswith('rpms_')
+
+    - name: Set Facts - current job file names
+      set_fact:
+        _currentJobFileNames: "{{ _currentJobFileNames + [item] }}"
```
Comment (rtamalin) [inline scripts/jenkins/cloud/ansible/roles/list_packages/tasks/packages_from_old_jobs.yml:93]: As above for _currentJobFileNames

File: scripts/jenkins/cloud/ansible/roles/list_packages/tasks/packages_from_old_jobs.yml (hunk 1 above)
Comment (djz88) [reply scripts/jenkins/cloud/ansible/roles/list_packages/tasks/packages_from_old_jobs.yml:69]: should be removed, thx

File: scripts/jenkins/cloud/ansible/roles/list_packages/tasks/packages_from_old_jobs.yml (hunk 2 above)
Comment (djz88) [reply scripts/jenkins/cloud/ansible/roles/list_packages/tasks/packages_from_old_jobs.yml:83]: thx for heads up, I ll change that

File: scripts/jenkins/cloud/ansible/roles/list_packages/tasks/packages_from_old_jobs.yml (hunk 3 above)
Comment (djz88) [reply scripts/jenkins/cloud/ansible/roles/list_packages/tasks/packages_from_old_jobs.yml:93]: the same as above

**Review by rtamalin — APPROVED** [review]
LGTM

**Review by flaviodsr — APPROVED** [review]
LGTM, thanks!

**Pull request merged**